presto-main/src/test/java/io/prestosql/execution/TestCreateSchemaTask.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.Session;
+import io.prestosql.connector.CatalogName;
+import io.prestosql.metadata.AbstractMockMetadata;
+import io.prestosql.metadata.Catalog;
+import io.prestosql.metadata.CatalogManager;
+import io.prestosql.metadata.SchemaPropertyManager;
+import io.prestosql.security.AllowAllAccessControl;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.CatalogSchemaName;
+import io.prestosql.spi.security.PrestoPrincipal;
+import io.prestosql.sql.tree.CreateSchema;
+import io.prestosql.sql.tree.QualifiedName;
+import io.prestosql.transaction.TransactionManager;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.prestosql.spi.StandardErrorCode.ALREADY_EXISTS;
+import static io.prestosql.testing.TestingSession.createBogusTestingCatalog;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestCreateSchemaTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private Session testSession;
+    TestCreateSchemaTask.MockMetadata metadata;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Catalog testCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        catalogManager.registerCatalog(testCatalog);
+        SchemaPropertyManager schemaPropertyManager = new SchemaPropertyManager();
+        schemaPropertyManager.addProperties(testCatalog.getConnectorCatalogName(), ImmutableList.of());
+        testSession = testSessionBuilder()
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .build();
+        metadata = new TestCreateSchemaTask.MockMetadata(
+            schemaPropertyManager,
+            testCatalog.getConnectorCatalogName());
+    }
+
+    @Test(expectedExceptions = PrestoException.class)
+    public void testDuplicatedCreateSchema()
+    {
+        duplicatedCreateSchema("test_db", false);
```

Comment:
Please move the assertions to test methods. And be more specific where we expect an exception.
What about getting rid of `duplicateCreateSchema` and having `createSchema` method instead which just executes the statement.

Then this test method would look like this:
```
   createSchema("test_db", false);
   assertEquals(metadata.getCreateSchemaCount(), 1);
   assertThatExceptionOfType(PrestoException.class)
            .isThrownBy(() -> crateSchema("test_db", false))
            .withMessage(".....")
```
   
```